src/terminal/parser/stateMachine.cpp

```diff
@@ -1838,12 +1839,17 @@ void StateMachine::ProcessString(const std::wstring_view string)
     size_t start = 0;
     size_t current = start;
 
+    _currentString = string;
+    _runOffset = 0;
+    _runSize = 0;
+
     while (current < string.size())
     {
         // The run will be everything from the start INCLUDING the current one
         // in case we process the current character and it turns into a passthrough
         // fallback that picks up this _run inside `FlushToTerminal` above.
-        _run = string.substr(start, current - start + 1);

```

Comment:
Repeatedly `substr` turns out to be quite expensive and takes about 10%-12% of the total CPU.

Reply:
I honestly forgot whether the 10% here is in OpenConsole or WT, or both...

Reply:
I feel like you should leave the comment/detail somewhere in here that we shouldn't get the future-bright-idea to "improve safety" here by re-wrapping this into a convenience class. It needs to stay as a raw sort of offset thing because of how high performance this needs to be.